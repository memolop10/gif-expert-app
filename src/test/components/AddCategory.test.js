import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";


describe(' Test in <AddCategory/> ', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> )

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> )
    })

    test('should render correctly', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('should change the input', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola Mundo'

        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe( value )

    })
    
    test('should not post the information with submit', () => {
        
        wrapper.find('form').simulate('submit',{ preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled();
    })

    test('should call setCategories and clear text box', () => {
        
        const input = wrapper.find('input');
        const value = 'Soy Goku';
        
        input.simulate('change', { target: { value } } )

        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);

        //Que se haya llamado una vez y reciba un funcion de parametro
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );

        expect( input.prop('value') ).toEqual( '' ) 
        
    })
    
    
    
});
