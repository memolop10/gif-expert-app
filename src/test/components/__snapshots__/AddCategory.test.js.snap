// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Test in <AddCategory/>  should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
